src/DataSourceProvider.vue

```diff
@@ -1,19 +1,138 @@
 <template>
-  <div>
-    <h1>Your fliplet-widget-data-source-provider widget will be added here.</h1>
-    <First-Component></First-Component>
-  </div>
+  <section>
+    <div v-if="isLoading" class="spinner-holder animated">
+      <div class="spinner-overlay">Loading...</div>
+    </div>
+    <div v-else-if="hasError" class="alert alert-danger" role="alert">
+      {{ errorMessage }}
+    </div>
+    <div v-else>
+      <DataSourceSelector
+        :currentAppDataSources="appDataSources" :otherDataSources="allDataSources"
+        :selectedDataSource="selectedDataSource"
+        :changeDataSource="changeDataSource"
+        :showAll="showAll"
+        @selectedDataSourceId="selectedDataSourceId = $emit"
+        @onDataSourceCreate="createDS"
+        @onShowAll="(event) => { showAllDataSources(event) }"
+        @onDataSourceChange="changeDataSource = !changeDataSource"
```

Comment:
Why is the `changeDataSource` property exposed to outside of the `DataSourceSelector` component or set from outside of the `DataSourceSelector` component? I don't see why that's necessary when it's a behaviour internal to the `DataSourceSelector` component.

Reply:
If we not controlling it with the main component and declare it in the `DataSourceSelector` we will always overwrite it when we updating the component and therefore will be an issue with the checkbox display. That is why I managing it with the main component.

Reply:
> [...] we will always overwrite it when we updating the component

I don't understand what you're referring to here.

Reply:
@tonytlwu Sorry I might have explained it wrong.
The reason why I refused that `changeDataSource` was inside of the `DataSourceSelector` is that when we initiating a loading screen in the data source provider we removing all components in the `DataSourceProvider` and after loading is complete they being mounted as new. So if we declare `changeDataSource` within `DataSourceSelector` as 
```
data() {
  return {
    changeDataSource: false
  }
}
```
It will always stay false, and we won't be able to see another state of the `DataSourceSelector`. That is why I transfer it threw the props, that our main component could control it.

Reply:
This problem only seems to exist because you've placed the loading state outside of the `DataSourceSelector.vue` file. If the `DataSourceSelector.vue` file contains the loading state, then the file using the `DataSourceSelector` component doesn't need to know what state it's in.

Reply:
@YaroslavOvdii ☝️